### DISCUSSION TITLE
How should we assign specific tasks to people?

### DISCUSSION THREAD

**Discussion by bnord** [deepskies/DeepSZSim#54] [Ideas]
I think the major tasks right now are:

1. Test the SZ physics, which is primarily the Battaglia profile. Eve Is that correct? How can we break that down into discrete independent tests? Are  folks all working on the same thing? 
    2. I think we should make sure that the branches are well defined for each task. Antwine's branch named `Antwine` should probably be named for the task. @kbanker's branch should be name for a specific unit test or set of unit tests. 
3. reorganize the code, which includes
    4. having all the proper config files in the proper place
    5. having all the code pieces connected
    6. putting the right functions in the right files
    7. I'm taking this to be my assignment bnord+SamMcD
4. making some design decisions -- e.g., the mass units, the data sets we should compare to.

**Comment by bnord** [comment]
@evavagiakis I assigned you to a bunch of tasks/issues, because I wasn't sure who else we should assign to, but I think we should def spread those out.

**Comment by evavagiakis** [comment]
1) I have been coordinating the to-dos over the google doc and slack, with the most recent round of assignments on slack. I think assigning to-dos over slack and zoom on the coworks works best for me. What would you prefer? 
1.2) I agree the branches and folders need to have specific names, and have started that discussion with Antwine
2) I agree the code needs to be reorganized. Right now, the most up-to-date code I have been working with is in https://github.com/deepskies/SimSZ/tree/dev-massdist. Any code from other branches in SimSZ is likely stale.

I am working to catch up on the 40 issues/discussions and will do my best to respond to them, but since I am working over google docs, slack, and now github, I am finding this a bit overwhelming. I think keeping discussions on slack, and code-specific issues on github is how I'd prefer to proceed

**Reply by bnord** [reply]
We will eventually need to put things into issues. When should we do that?

**Reply by bnord** [reply]
the issue with slack is that we will lose all that history after 90 days.

**Reply by bnord** [reply]
Would you like me to copy-paste our discussions from slack into github discussions after we finish it in slack?

**Reply by evavagiakis** [reply]
I think the overarching goals for the students are largely issues assigned to me, and I'm not planning on assigning any students any code pushes, so I plan to assign students issues based on their own scripts

**Reply by bnord** [reply]
what do you mean here? `so I plan to assign students issues based on their own scripts`

**Reply by bnord** [reply]
Oh, I understand.
They will work locally and not yet push back to the repo themselves.

How will the code get into the repo then?

**Comment by bnord** [comment]
Sorry for the multiple platforms. Since most of what we're doing right now is code dev, I thought issues would be the right place to assign people things.

**Reply by bnord** [reply]
another concern with slack/google docs is that we may need to have multiple threads of questions / discussion open concurrently.

**Reply by bnord** [reply]
It's almost impossible to do that in slack and not lose track of a given thread.
I could imagine it working in a google doc. Would you like me to try to move some things there?

**Comment by evavagiakis** [comment]
One thing that will help is using github actions like email or wiki page updates rather than slack; I get a new email for every message sent, meaning I just got 7 emails from this single discussion thread, and it's taking me a lot of time to go through them. I don't want to change my email preferences or I'll lose track of what has been updated. Can we keep discussion to slack or zoom, and then make an issue with succinct, assigned tasks for a given week, for example? So we use github for action item tracking (along with any pertinent notes from the discussion). I'd like to not use github for short messages or thoughts.

**Reply by bnord** [reply]
maybe I'm using the @ thing wrong. I'll try delete that.
